### PR TITLE
[master < T0830-MG] Fix SHOW REPLICATION ROLE and SHOW REPLICAS

### PR DESCRIPTION
**Pull request by gitbuda**
- [x] Write E2E tests
- [x] Check, and update documentation if necessary
- [x] Update [changelog](https://docs.memgraph.com/memgraph/changelog) -> https://github.com/memgraph/docs/pull/381

### CODE REVIEW

File: tests/e2e/replication/show.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import mgclient
+from common import execute_and_fetch_all
+
+
+def test_show_replication_role(connection):
+    cursor = connection.cursor()
+    data = execute_and_fetch_all(cursor, "SHOW REPLICATION ROLE;")
+
+    assert cursor.description[0].name == "replication role"
+    assert data[0][0] == "main"
```
Comment (antaljanosbenjamin) [inline tests/e2e/replication/show.py:26]: Maybe make a connection to a replica and check their role also?

**Review by antaljanosbenjamin — COMMENTED** [review]
Could you please also make a small pr to mention the renamed header in the changelog? I think the result of the query is not mentioned anywhere in the docs, so from that perspective we are good to go.

File: tests/e2e/replication/show.py (hunk 1 above)
Comment (gitbuda) [reply tests/e2e/replication/show.py]: Done please check it out again 😄

**Comment by gitbuda** [comment]
@antaljanosbenjamin and @kostasrim I consider this done, feel free to review it 😄

Adding @BorisTasevski because he is doing some work on the side of the tests as well (FYI type of thing) 😄

**Comment by gitbuda** [comment]
Closes #374 and #375

File: tests/e2e/replication/show.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import typing
```
Comment (jbajic) [inline tests/e2e/replication/show.py:12]: You are not using typing

File: tests/e2e/replication/show.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    assert cursor.description[0].name == "replication role"
+    assert data[0][0] == role
+
+
+def test_show_replicas(connection):
+    cursor = connection(7687, "main").cursor()
+    actual_data = execute_and_fetch_all(cursor, "SHOW REPLICAS")
+
+    expected_column_names = ["name", "socket_address", "sync_mode", "timeout"]
+    actual_column_names = list(map(lambda x: x.name, cursor.description))
```
Comment (jbajic) [inline tests/e2e/replication/show.py:37]:
```suggestion
    actual_column_names = [x.name for x in cursor.description]
```

File: tests/e2e/replication/show.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    for expected, actual in zip_longest(expected_column_names, actual_column_names):
+        assert expected == actual
+
+    expecte_data = (
+        ("replica_1", "127.0.0.1:10001", "sync", 0),
+        ("replica_2", "127.0.0.1:10002", "sync", 1.0),
+        ("replica_3", "127.0.0.1:10003", "async", None),
+    )
+    for expected, actual in zip_longest(expecte_data, actual_data):
+        assert expected == actual
```
Comment (jbajic) [inline tests/e2e/replication/show.py:47]:
```suggestion
    expected_data = (
        ("replica_1", "127.0.0.1:10001", "sync", 0),
        ("replica_2", "127.0.0.1:10002", "sync", 1.0),
        ("replica_3", "127.0.0.1:10003", "async", None),
    )
    for expected, actual in zip_longest(expected_data, actual_data):
        assert expected == actual
```

**Review by jbajic — CHANGES_REQUESTED** [review]

File: tests/e2e/replication/conftest.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import pytest
+
+from common import execute_and_fetch_all, connect
+
+
+# The fixture here is more complex because the connection has to be
+# parametrized based on the test parameters (info has to be available on both
```
Comment (kostasrim) [inline tests/e2e/replication/conftest.py:18]:
```suggestion
# parameterized based on the test parameters (info has to be available on both
```

File: tests/e2e/replication/show.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def test_show_replicas(connection):
+    cursor = connection(7687, "main").cursor()
+    actual_data = execute_and_fetch_all(cursor, "SHOW REPLICAS")
+
+    expected_column_names = ["name", "socket_address", "sync_mode", "timeout"]
+    actual_column_names = list(map(lambda x: x.name, cursor.description))
+    for expected, actual in zip_longest(expected_column_names, actual_column_names):
+        assert expected == actual
```
[the remark this answers is not in the document]
Comment (gitbuda) [reply tests/e2e/replication/show.py]: Nice idea!

**Comment by gitbuda** [comment]
Sorry for the silly mistakes, somehow I have to put Grammarly into my workflow... @jbajic please check again 😄

**Review by jbajic — APPROVED** [review]

**Review by kostasrim — APPROVED** [review]

**Pull request merged**